civis/io/_tables.py

```diff
@@ -671,6 +670,23 @@ def dataframe_to_civis(df, database, table, api_key=None, client=None,
     credential_id : str or int, optional
         The ID of the database credential.  If ``None``, the default
         credential will be used.
+    primary_keys: list[str], optional
```

Comment:
Actually, this is relevant regardless of databases for imports in `upsert` mode. Our Platform-side upsert logic requires primary keys (in order to identify common records between staging and final tables) and last-modified columns (to determine which record is newer). Happy to talk more offline about this.

Reply:
In that case, I would call that out explicitly in the documentation, because it won't necessarily be intuitive to a user who's used to redshift.

Reply:
Both of those parameters have a note in their section that says exactly that. It is also listed in our [API Documentation for the `imports/files/csv` endpoint](https://platform.civisanalytics.com/spa/#/api#post_imports_files_csv) and [Our Helpdesk docs for new imports under "Upsert Mode"](https://civis.zendesk.com/hc/en-us/articles/360027016851-Advanced-CSV-Imports-via-the-Civis-API). I can add a note to the `existing_table_rows` documentation as well, or add some RST emphasis to draw extra attention to it - let me know what you think.

Reply:
Open to suggestions here but the challenge appears to be can we educate users on what an "upsert" is in a docstring? The primary keys are relevant regardless of the destination database because they are required to generate the SQL for an upsert operation.

One change we could consider to this description is: "A list of the primary key column(s) in the destination table _**that uniquely identify a record**_."

Reply:
Building off of that, I might say "A list of the primary key column(s) of the destination table that uniquely identify a record. If existing_table_rows is "upsert", this field is required. Note that this is true regardless of whether the destination database itself requires a primary key." This might seem like overkill, but because "primary key" is somewhat overloaded, I think it's useful to be as explicit as possible.

Reply:
Added more description to primary keys where relevant.